src/Confluent.Kafka/Consumer.cs

```diff
@@ -408,6 +408,11 @@ private void LogCallback(IntPtr rk, int level, string fac, string buf)
         /// </remarks>
         public Consumer(IEnumerable<KeyValuePair<string, object>> config)
         {
+            if (config.FirstOrDefault(prop => prop.Key == "group.id").Value == null)
```

Comment:
I think these configuration keys should be compared using `string.Equals(x, y, StringComparison.Ordinal)` as recommended [here](https://msdn.microsoft.com/en-US/library/dd465121(v=vs.110).aspx#choosing_a_stringcomparison_member_for_your_method_call).